Automattic/pocket-casts-android
# Patron: Upgrade screen UI adjustments

## Description

This updates the new upgrade screen to keep the upgrade button position fixed (pdeCcb-2vS-p2#comment-1989), updates the pager height to the tallest page, and fixes jumping of the pager due to the variable header at the top in large font sizes.

## Testing Instructions
1. Apply [this patch](https://github.com/Automattic/pocket-casts-android/files/11343238/upgrade-testing-patch-with-patron.patch)
2. Increase device font size
3. Run the app being logged out
4. Subscribe to any podcast
5. Go to Podcasts
6. Tap the folder icon
7. Notice that 
    - the upgrade screen is scrollable with the upgrade button fixed
    - switching between the pages does not create a jumping effect
    - page sizes are the same

## Screenshots or Screencast 

https://user-images.githubusercontent.com/1405144/235459006-38300d60-3dbb-4d1a-aaab-3eac42c87e95.mp4


## Checklist
 
#### I have tested any UI changes...
